packages/playground/src/views/settings.vue

```diff
@@ -0,0 +1,245 @@
+            type="number"
+            v-model="selectedTimeout"
+          ></v-text-field>
+        </input-validator>
+        <v-card-actions class="justify-end">
+          <v-btn :disabled="!isValidTimeout" @click="UpdateTimeout" class="justify-end ml-auto"
+            >Update</v-btn
+          ></v-card-actions
+        >
+      </form-validator>
+    </v-card>
+  </view-layout>
+</template>
+<script lang="ts">
+import md5 from "md5";
+import { ref, watch } from "vue";
+import { useTheme } from "vuetify";
+
+import { createCustomToast, ToastType } from "@/utils/custom_toast";
+
+import { getCredentials, updateCredentials } from "../utils/credentials";
+export default {
+  name: "Settings",
+  setup() {
+    const theme = useTheme();
+    const KEY = "APP_CURRENT_THEME";
+
+    const themes = ["Dark Mode", "Light Mode"];
+
+    const currentTheme = localStorage.getItem(KEY);
+
+    watch(theme.global.name, theme => localStorage.setItem(KEY, theme));
+
+    const selectedTheme = ref(currentTheme == "light" ? "Light Mode" : "Dark Mode");
+
+    const currentPassword = ref("");
+    const newPassword = ref("");
+    const confirmPassword = ref("");
+    const selectedTimeout = ref(window.env.TIMEOUT / 1000);
+    const isValidTimeout = ref(false);
+    const isValidPassword = ref(false);
+    const isValidTheme = ref(false);
+    function validateTheme() {
+      if (selectedTheme.value.split(" ")[0].toLowerCase() == currentTheme) {
+        return { message: "Select a theme different from the current one." };
+      }
+    }
+
+    function UpdateTheme() {
+      localStorage.setItem(KEY, selectedTheme.value == "Light Mode" ? "light" : "dark");
+      theme.global.name.value = selectedTheme.value == "Light Mode" ? "light" : "dark";
+    }
+    function validateCurrentPassword() {
+      if (getCredentials().passwordHash !== md5(currentPassword.value)) {
+        return { message: "Incorrect Password." };
+      }
+    }
+    /**
+     * Checks that new password isn't the same as the current one.
+     */
+    function validateNewPassword() {
+      if (newPassword.value === currentPassword.value) {
+        return { message: "New password cannot be the same as current password. Please enter a different password." };
+      }
+    }
+    function validateConfirmPassword(value: string) {
+      if (value !== newPassword.value) {
+        return { message: "Passwords should match." };
+      }
+    }
+    /** Updates user credentials with the hashes produced by the new password  */
+    async function UpdatePassword() {
+      await updateCredentials(currentPassword.value, newPassword.value);
+
+      createCustomToast("Password Updated!", ToastType.success);
+      // reset input fields
+      currentPassword.value = "";
+      newPassword.value = "";
+      confirmPassword.value = "";
+      isValidPassword.value = false;
+    }
+    function UpdateTimeout() {
+      try {
+        window.env.TIMEOUT = selectedTimeout.value * 1000;
```

Comment:
What is this Timeout used for? request? should we update all requests with this timeout?

Reply:
made timeout persistent in local storage to be applicable session wide